test/gpx-image.test.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * GPX Image Generation Test
+ *
+ * This test verifies that the GPX image generation produces consistent results.
+ * It generates an image for a known tour and compares it with a reference image.
+ */
+
+import puppeteer from "puppeteer";
+import fs from "fs-extra";
+import path from "path";
+import sharp from "sharp";
+import crypto from "crypto";
+
+// Test configuration
+const TEST_TOUR_ID = 28308;
+const REFERENCE_IMAGE_PATH = path.join(__dirname, "fixtures/gpx_image_reference.webp");
+const GENERATED_IMAGE_PATH = path.join(__dirname, "fixtures/gpx_image_generated.webp");
+const TEMP_PNG_PATH = path.join(__dirname, "fixtures/gpx_image_temp.png");
+
+// Same settings as in gpxUtils.js
+const VIEWPORT = { width: 1200, height: 800 };
+const GPX_URL = `http://localhost:8080/public/gpx/08/${TEST_TOUR_ID}.gpx`;
+const HEADLESS_URL = `http://localhost:8080/public/headless-leaflet/index.html?gpx=${GPX_URL}`;
```

Comment:
This test is failing at the moment because the URL http://localhost:8080/public/headless-leaflet/index.html yields nothing. What is the intention here?

```suggestion
const GPX_URL = `https://www.zuugle.at/public/gpx/08/${TEST_TOUR_ID}.gpx`;
const HEADLESS_URL = `https://www.zuugle.at/public/headless-leaflet/index.html?gpx=${GPX_URL}`;
```